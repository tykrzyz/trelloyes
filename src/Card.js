import './Card.css'

function Card(props){
    return(
        <div className='Card'>
            <button type='button' onClick={e => props.delete(props.id)}>
                delete
            </button>
            <h3>{props.title}</h3>
            <p>{props.content}</p>
        </div>
    );
}

export default Card