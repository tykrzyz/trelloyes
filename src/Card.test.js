import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import renderer from 'react-test-renderer';

describe('<Card />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<Card title='Hello' content='World'/>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
        });
})