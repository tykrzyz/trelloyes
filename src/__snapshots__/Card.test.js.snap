// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    Hello
  </h3>
  <p>
    World
  </p>
</div>
`;
